pkg/services/notary/notary.go

```diff
@@ -316,7 +317,28 @@ func (n *Notary) finalize(tx *transaction.Transaction) error {
 			break
 		}
 	}
-	return n.onTransaction(tx)
+	newTx, err := updateTxSize(tx)
+	if err != nil {
+		return fmt.Errorf("failed to update completed transaction's size: %w", err)
+	}
+
+	return n.onTransaction(newTx)
+}
+
+// updateTxSize returns transaction with re-calculated size and an error.
+func updateTxSize(tx *transaction.Transaction) (*transaction.Transaction, error) {
+	bw := io.NewBufBinWriter()
+	tx.EncodeBinary(bw.BinWriter)
+	if bw.Err != nil {
+		return nil, fmt.Errorf("encode binary: %w", bw.Err)
+	}
+	newTx := &transaction.Transaction{Network: tx.Network}
+	br := io.NewBinReaderFromBuf(bw.Bytes())
+	newTx.DecodeBinary(br)
```

Comment:
Is encoding + decoding really necessary? Why don't have helper function or method inside `transaction` package?

Reply:
My fear is that exposing these internals (providing `(*Transaction) RecalcSize()` may open a can of worms worse than this encoding/decoding. And to properly calculate size we need to serialize the transaction anyway.

Reply:
We may create `(*transaction).UpdateSize()`  method, but the notary module is the only place where it can be applied.